Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -724,301 +732,209 @@ public int GetStateValue(InteractableStates.InteractableStateEnum state)
+            IsEnabled = true;
+            HasObservation = false;
+            HasObservationTargeted = false;
+            IsInteractive = false;
+            IsTargeted = false;
+            IsToggled = false;
+            IsVisited = false;
+            HasCollision = false;
+            HasCustom = false;
         }
 
-        /// <summary>
-        /// Change the visited state
-        /// </summary>
-        public virtual void SetVisited(bool visited)
-        {
-            IsVisited = visited;
-            SetState(InteractableStates.InteractableStateEnum.Visited, visited);
-        }
+        #endregion State Utilities
+
+        #region Dimensions Utilities
 
         /// <summary>
-        /// Change the toggled state
+        /// Increases the Current Dimension by 1. If at end (i.e Dimensions - 1), then loop around to beginning (i.e 0)
         /// </summary>
-        public virtual void SetToggled(bool toggled)
+        public void IncreaseDimension()
         {
-            SetState(InteractableStates.InteractableStateEnum.Toggled, toggled);
-
-            // if in toggle mode
-            if (IsToggleButton)
+            if (CurrentDimension == Dimensions - 1)
             {
-                SetDimensionIndex(toggled ? 1 : 0);
+                CurrentDimension = 0;
             }
             else
             {
-                int selectedMode = Mathf.Clamp(Dimensions, 1, 3);
-                Debug.Log("SetToggled(bool) called, but SelectionMode is set to " + (SelectionModes)(selectedMode - 1) + ", so DimensionIndex was unchanged.");
+                CurrentDimension++;
             }
         }
 
         /// <summary>
-        /// Change the gesture state
+        /// Decreases the Current Dimension by 1. If at zero, then loop around to end (i.e Dimensions - 1)
         /// </summary>
-        public virtual void SetGesture(bool gesture)
+        public void DecreaseDimension()
         {
-            HasGesture = gesture;
-            SetState(InteractableStates.InteractableStateEnum.Gesture, gesture);
+            if (CurrentDimension == 0)
+            {
+                CurrentDimension = Dimensions - 1;
+            }
+            else
+            {
+                CurrentDimension--;
+            }
         }
 
         /// <summary>
-        /// Change the gesture max state
+        /// Helper method to convert number of dimensions to the appropriate SelectionModes
         /// </summary>
-        public virtual void SetGestureMax(bool gesture)
+        /// <param name="dimensions">number of dimensions</param>
+        /// <returns>SelectionModes for corresponding number of dimensions</returns>
+        public static SelectionModes ConvertToSelectionMode(int dimensions)
         {
-            HasGestureMax = gesture;
-            SetState(InteractableStates.InteractableStateEnum.GestureMax, gesture);
+            if (dimensions <= 0)
+            {
```

Comment:
nit: could be a switch case

Reply:
I actually don't like switch cases for short lists and it's not part of the coding guidelines